src/main/java/uk/gov/hmcts/ccd/domain/service/casefileview/CategoriesAndDocumentsService.java

```diff
@@ -175,9 +179,16 @@ private Document buildDocument(final String attributePath, @NonNull final Map<St
     }
 
     String resolveDocumentCategory(final String categoryOnDocument,
-                                   final String categoryOnFieldDefinition) {
-        return Optional.ofNullable(categoryOnDocument)
-            .orElseGet(() -> resolveDocumentCategory(categoryOnFieldDefinition));
+                                   final String categoryOnFieldDefinition,
+                                   final List<CategoryDefinition> categories) {
+        if (categoryOnDocument != null && !categories.stream()
+            .anyMatch(category ->
+                category.getCategoryId().equals(categoryOnDocument))) {
+            return UNCATEGORISED_KEY;
+        } else {
+            return Optional.ofNullable(categoryOnDocument)
+                .orElseGet(() -> resolveDocumentCategory(categoryOnFieldDefinition));
+        }
```

Comment:
```suggestion
        boolean isDocumentCategoryPresent = categoryOnDocument != null &&
            categories.stream()
                .anyMatch(category -> category.getCategoryId().equals(categoryOnDocument));

        return isDocumentCategoryPresent ? categoryOnDocument : resolveDocumentCategory(categoryOnFieldDefinition);
```
Please check this out. I think these makes the code more maintainable and easier to understand.